### DISCUSSION TITLE
'yarn storybook' creates node_modules/.cache in Yarn 2 PnP project

### DISCUSSION THREAD

**Discussion by turtle601** [storybookjs/storybook#25986] [General]
### Describe the bug

My project is using Yarn 2 PnP and consequently there exists no node_modules folder.

After I installed StoryBook with yarn version
```
yarn workspace @package/ui dlx storybook init
```

then run storybook

```
yarn workspace @package/ui storybook
```

them node_modules will be created.

Storybook had the [same issue](https://arc.net/l/quote/efhvyvcg) 3 years ago.

I am still curious whether the node_modules library is still unconditionally required to use storybook, 

or if there is a way to run it in pnp mode without creating node_modules.

**Comment by valentinpalkovic** [comment]
For convenience reasons, Storybook's cache directory is located in the node_modules folder, since it is normally not checked in via version control. I understand, that in yarn PnP mode this folder doesn't exist. When Storybook creates a node_modules/.cache folder, do you encounter any side effects? Does it have a negative impact in yarn pnp mode when the node_modules folder is created or is it just nasty to have it in place, although it isn't required?

**Reply by turtle601** [reply]
👋 Thank you for answering 👋

We suprised that it is nasty to have it in place, but storybook ran smoothly. 
(I haven't found any other problems yet.)

This may be a very surprising situation for users who are using Storybook and yarn berry pnp mode for the first time.

<img width="806" alt="스크린샷 2024-02-11 오후 6 35 02" src="https://github.com/storybookjs/storybook/assets/78203399/7d0955c4-7df9-4dd8-8bb6-8a5ceb2b9fda">

<br />
I felt that the command in the photo above especially supports yarn pnp mode.
It would be good to mention in the official document that pnp mode is supported, but the node_modules created there are essential.

**Reply by valentinpalkovic** [reply]
That's a good point. I will mention @jonniebigodes here to bring it onto his radar and to adjust the docs. 

Thank you for reaching out!

**Reply by turtle601** [reply]
@valentinpalkovic 
I wrote a message explaining that node_modules are created when running storybook in pnp mode.

Please give me a feedback for message about PR 👍

[Issue link](https://github.com/storybookjs/storybook/issues/25987)
[PR link](https://github.com/storybookjs/storybook/pull/25988)
